### ISSUE TITLE
Debug does not work on big projects (mixing js and typescript)

### ISSUE THREAD

**Issue by danielo515** [Microsoft/vscode#20118]
- VSCode Version: 1.9.0
- OS Version: Mac OS-X Sierra

The debug interface is unresponsive on a big project. With unresponsive I mean that it does not do anything at all: it does not throw any error, it is not able to generate any launch file, it does not start the debugging, it is unable to connect to an existing node debugger...

I'm afraid that this is related to the fact that I'm mixing some js modules with typescript modules. I'm using a monorepo structure, and I open the entire monorepo folder with vscode.

- Root
  - .vscode
  - package.json (global)
  - src
    - jsmodule
      - package.json
    - jsmodule
      - package.json
    - tsmodule
      - .vscode
      - package.json
      - tsconfig.json

Before adding the typescript module everything was working fine, that's why I think that this is related to the fact that I have a typescript module inside the folder structure. 

If there is any way of extracting any extra log information or traces I would love to provide them.

Regards

**EDIT**

I think I found the problem. Two things seems to be involved:
- The existence of a `.vscode` folder under the typescript module subfolder
- My old launch.json seems to be wrong

**Comment by isidorn** [comment]
Well if you launch.json seems to be wrong then that is probably the cause of the issue.
If there is a debugging issue here please try to provide steps and share your launch.json
Having a `.vscode` folder underf some subfolder should not affect the behvaior of vscode

As far as I understand you have found and solved the issue thus closing this unless more data is provided

**Comment by danielo515** [comment]
Dear @isidorn ,

Yes, I have solved the issue deleting my old file and generating a new one. Then, I copied some debug options back to the newly created launch.json and it still working. I'll continue adding configurations until I find the problematic one.

That said, I really think there should be some incompatibility between the latest version of VS code and the previous one, because that same file was working on 1.8.1. 

Regards
